### PR TITLE
#10339 Legend support for ArcGIS MapServer/ImageServer layers

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

This PR introduces legend support for ArcGIS MapServer/ImageServer layers

image

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Minor changes to existing features

## Issue

**What is the current behavior?**

#10339

**What is the new behavior?**

The arcgis layers will show a legend inside the TOC

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

### CODE REVIEW

**Review by offtherailz — COMMENTED** [review]
Looks to be working.
- [ ] Printing legend do not seems to be shown. Is it intentional?
- [x] (one comment inline about doc change)

File: docs/developer-guide/maps-configuration.md [hunk 1]
```diff
@@ -1316,7 +1316,10 @@ We have two type of configuration, the first one allow to render only a single l
 {
     "type": "arcgis",
     "name": "0",
-    "url": "https://arcgis-example/rest/services/MyService/MapServer"
+    "url": "https://arcgis-example/rest/services/MyService/MapServer",
+    "options": {
+      "layers": [{ "id": 0 }, { "id": 1 }]
+    },
```
[the remark this answers is not in the document]
Comment (allyoucanmap) [reply docs/developer-guide/maps-configuration.md:1322]: It's not mandatory, but some sub layers could not display the legend or identify response.
I discover that some sublayers could behave like groups so they need information about the hierarchy to get the correct layers in the legend.

[the remark this answers is not in the document]
Comment (offtherailz) [reply docs/developer-guide/maps-configuration.md:1322]: Ok. In fact is strongly raccommended to have it, for queriable and legend. 
So I'd clarify better the roles of the attributes. I should explain it like this ( replacing the actual explaination) :

````
[...]
An ArcGIS MapServer source is a composition of different layers to create a map. The layer is identified by the `arcgis` type, containing `url` and `options.layers` properties . e.g. 

```json
{
    "type": "arcgis",
    "url": "https://arcgis-example/rest/services/MyService/MapServer",
    "options": {
      "layers": [{ "id": 0 }, { "id": 1 }]
    },
    "title": "Title",
    "group": "",
    "visibility": true,
    "queriable": true
}

Where: 

- `url` is the URL of the MapServer source. 
- `options.layers` is the list of object containing the ids of the layers. Required to support `queriable` option and legend support.
- `name` (optional). When present, the MapStore layer will show only the layer with the id specified in the `name` attribute. e.g. 
- `queriable` 

```json
{
    "type": "arcgis",
    "name": "0",
    "url": "https://arcgis-example/rest/services/MyService/MapServer"
    "url": "https://arcgis-example/rest/services/MyService/MapServer",
    "options": {
      "layers": [{ "id": 0 }, { "id": 1 }]
    },
    "title": "Title",
    "group": "",
    "visibility": true,
    "queriable": true
}
```

````

**Comment by tdipisa** [comment]
@allyoucanmap (FYI @offtherailz) 

>@tdipisa how should we manage this, separate issue?

yes please, remember to assign the label ArcGIS and put it to 2024,02.00 milestone for the moment.

File: docs/developer-guide/maps-configuration.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (allyoucanmap) [reply docs/developer-guide/maps-configuration.md:1320]: @offtherailz I just updated the documentation in this commit 848cdcdc03f9acf115cf9fd84b2df3e9568f3f1c

**Comment by allyoucanmap** [comment]
@tdipisa here the new issue https://github.com/geosolutions-it/MapStore2/issues/10411

**Review by offtherailz — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@ElenaGallo please test this enhancement on dev, thanks

**Pull request merged**